### PR TITLE
BUG: .loc with MultiIndex with names[1] = 0

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -1552,12 +1552,19 @@ def droplevel(self, level=0):
 
         levnums = sorted(self._get_level_number(lev) for lev in level)[::-1]
 
-        if len(level) == 0:
+        return self._drop_level_nums(levnums)
+
+    def _drop_level_nums(self, levnums: List[int]):
```
Comment (jreback) [inline pandas/core/indexes/base.py:1557]: this returns an Index/MultiIndex right? (I know this code is all convoluted, but if can type pls do)

Comment (jreback) [inline pandas/core/indexes/base.py:1557]: rename to _drop_level_number

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/base.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexes/base.py:1557]: tried that, mypy complained.  some Index methods need to be moved to MultiIndex to get tight annotations

Comment (jbrockmendel) [reply pandas/core/indexes/base.py:1557]: sure

Comment (jbrockmendel) [reply pandas/core/indexes/base.py]: updated+green

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -1552,12 +1552,19 @@ def droplevel(self, level=0):
 
         levnums = sorted(self._get_level_number(lev) for lev in level)[::-1]
 
-        if len(level) == 0:
+        return self._drop_level_numbers(levnums)
+
+    def _drop_level_numbers(self, levnums: List[int]):
```
Comment (jreback) [inline pandas/core/indexes/base.py:1557]: sorry to be pedantic, can you make this _drop_level_number (everything else is singular)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/base.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/indexes/base.py:1557]: it specifically takes `List[int]`

**Review by jreback — APPROVED** [review]

**Pull request merged**